gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageOptions.java

```diff
@@ -181,6 +182,41 @@ public void throwIfNotValid() {
     checkArgument(!isNullOrEmpty(getAppName()), "appName must not be null or empty");
   }
 
+  @Override
+  public String toString() {
+    return "CredentialOptions{\n"
+        + ("grpcEnabled: " + isGrpcEnabled() + '\n')
+        + ("rootUrl: " + getStorageRootUrl() + '\n')
+        + ("projectId: " + getProjectId() + '\n')
+        + ("appName: " + getAppName() + '\n')
+        + ("autoRepair: " + isAutoRepairImplicitDirectoriesEnabled() + '\n')
+        + ("inferImplicitDirectories: " + isInferImplicitDirectoriesEnabled() + '\n')
+        + ("durationMillis: " + getMaxWaitMillisForEmptyObjectCreation() + '\n')
+        + ("maxListItemsPerCall: " + getMaxListItemsPerCall() + '\n')
+        + ("maxRequestsPerBatch: " + getMaxRequestsPerBatch() + '\n')
+        + ("batchThreads: " + getBatchThreads() + '\n')
+        + ("maxHttpRequestRetries: " + getMaxHttpRequestRetries() + '\n')
+        + ("httpRequestConnectTimeout: " + getHttpRequestConnectTimeout() + '\n')
+        + ("httpRequestReadTimeout: " + getHttpRequestReadTimeout() + '\n')
+        + ("transportType: " + getTransportType() + '\n')
+        + ("proxyAddress: " + getProxyAddress() + '\n')
+        + ("proxyUsername: " + toSecretString(getProxyUsername()) + '\n')
```

Comment:
While we did it this way in the past, I think it will be better to invest to implementing [less error-prone and more maintainable solution](https://github.com/google/auto/issues/132#issuecomment-54777277).

Reply:
Done.